src/components/Search.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+
+interface Props {
+  intitialSearchText: string;
+  onSearch: (searchText: string) => void;
+}
+export default function Search({ intitialSearchText, onSearch }: Props) {
+  const [searchText, setSearchText] = useState(intitialSearchText);
+
+  return (
+    <div>
```

Comment:
Input + button = semanticaly this is a form. And html form will automaticaly handle the enter press case in 'onSubmit' handler (you will not need the code on Line 16)